core/tx_pool.go

```diff
@@ -301,12 +301,11 @@ func (pool *TxPool) loop() {
 		case ev := <-pool.chainHeadCh:
 			if ev.Block != nil {
 				pool.mu.Lock()
-				//if pool.chainconfig.IsHomestead(ev.Block.Number()) {
-				//	pool.homestead = true
-				//}
-				//pool.reset(head.Header(), ev.Block.Header())
-				//head = ev.Block
-
+				if pool.chainconfig.IsS3(ev.Block.Epoch()) {
```

Comment:
which signer shall we use?

Reply:
S3 is the condition we used to include homestead when we did the upgrade before.